spec/realtime/failure.test.js

```diff
@@ -333,5 +333,31 @@ define(['ably', 'shared_helper', 'async'], function(Ably, helper, async) {
 		80017
 	);
 
+	exports.idle_transport_timeout = function(test) {
+		var realtime = helper.AblyRealtime({realtimeRequestTimeout: 100}),
+			originalOnProtocolMessage;
+
+		test.expect(3);
+
+		realtime.connection.connectionManager.on('transport.pending', function(transport) {
+			originalOnProtocolMessage = transport.onProtocolMessage;
+			transport.onProtocolMessage = function(message) {
+				if(message.action === 4) {
+					message.connectionDetails.maxIdleInterval = 100;
+				}
+				originalOnProtocolMessage.call(this, message);
+			};
+		});
+
+		realtime.connection.once('connected', function() {
+			realtime.connection.once(function(statechange) {
+				test.equal(statechange.current, 'disconnected', 'check connection goes to disconnected');
+				test.equal(statechange.reason.code, 80003, 'check code');
+				test.equal(statechange.reason.statusCode, 408, 'check statusCode');
+				closeAndFinish(test, realtime);
+			});
+		});
```

Comment:
Ideally sure, but I'd rather not add an ably-js test that takes 15s minimum to complete, (been trying to rewrite tests that require waits, not add more, test suite takes too long as it is). Might be a thing to add if in the future we add a connectparam that allows the lib to request a custom heartbeat period from realtime. (Anyway, if realtime stops sending heartbeats we'll know soon enough, given ably-js now fails the transport if it doesn't get a heartbeat...)